web-components/collapse/collapse.svelte

```diff
@@ -0,0 +1,181 @@
+  const toggle = (e) => {
+    e.preventDefault()
+
+    // We only support toggling if this component isn't being 'controlled'
+    const toggleTo = !isOpenInternal
+    if (isOpen === undefined) isOpenInternal = toggleTo
+    dispatcher('toggle', { open: toggleTo })
+  }
+</script>
+
+<details class="leoCollapse" open={isOpenInternal}>
+  <summary on:click={toggle}>
+    {#if $$slots.icon}
+      <div class="icon">
+        <slot name="icon" />
+      </div>
+    {/if}
+    <div class="title">
+      {title}
+    </div>
+    <div class="arrow">
+      <svg
+        width="16"
+        height="9"
+        viewBox="0 0 16 9"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <path
+          fill-rule="evenodd"
+          clip-rule="evenodd"
+          d="M0.234303 0.234362C0.546723 -0.0780572 1.05325 -0.0780572 1.36567 0.234362L7.79999 6.66868L14.2343 0.234363C14.5467 -0.078056 15.0533 -0.0780559 15.3657 0.234363C15.6781 0.546783 15.6781 1.05331 15.3657 1.36573L8.64852 8.08289C8.17989 8.55152 7.42009 8.55152 6.95146 8.08289L0.234303 1.36573C-0.0781163 1.05331 -0.0781163 0.546782 0.234303 0.234362Z"
+          fill="currentColor"
+        />
+      </svg>
+    </div>
+  </summary>
+  <div class="content">
+    <slot />
+  </div>
+</details>
+
+<style lang="scss">
+  .leoCollapse {
+    --background-color: var(
+      --leo-collapse-background-color,
+      var(--color-container-background)
+    );
+    --background-color-hover: var(
+      --leo-collapse-background-color-hover,
+      var(--color-container-interactive-background)
+    );
+    --icon-size: var(--leo-collapse-icon-size, 24px);
+    --transition-duration: var(--leo-collapse-transition-duration, 0.12s);
+    --border-color: var(
+      --leo-collapse-border-color,
+      var(--color-divider-subtle)
+    );
+    --border-color-hover: var(
+      --leo-collapse-border-color-hover,
+      var(--color-primary-20)
+    );
+    --shadow: var(--leo-collapse-shadow, var(--effect-elevation-01));
+    --shadow-hover: var(
+      --leo-collapse-shadow-hover,
+      var(--effect-elevation-02)
+    );
+    --shadow-focus: var(--leo-collapse-shadow-focus, var(--effect-focus-state));
+    --radius: var(--leo-collapse-radius, var(--radius-16));
+    --summary-color: var(
+      --leo-collapse-summary-color,
+      var(--color-text-primary)
+    );
+    --summary-color-hover: var(
+      --leo-collapse-summary-color,
+      var(--color-text-interactive)
+    );
+    --icon-color: var(--leo-collapse-icon-color, var(--color-icon-default));
+    --icon-color-hover: var(
+      --leo-collapse-icon-color-hover,
+      var(--summary-color-hover)
+    );
+
+    @media (prefers-reduced-motion) {
+      --transition-duration: 0s;
+    }
+
+    background-color: var(--background-color);
+    box-shadow: var(--shadow);
+    border-radius: var(--radius);
+    border: 1px solid var(--border-color);
+    transition: box-shadow var(--transition-duration) ease-in-out,
```

Comment:
@fallaciousreasoning, can we include `color` in the transition so that the title transitions along with the background on hover?

Reply:
:+1: good call. Turns out I'd tried to do this but written `--var(--transition-duration)` which, believe it or not, is not how you're meant to do it :laughing:

Reply:
It's so buttery smooth! Thank you!